ducktape/cluster/ansible.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2022 Redpanda Data Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Any, Optional
+from ducktape.cluster.json import JsonCluster
+from ansible.inventory.manager import InventoryManager
+from ansible.parsing.dataloader import DataLoader
+from ansible.vars.manager import VariableManager
+
+import os
+import json
+
+
+class AnsibleCluster(JsonCluster):
+    """
+    An implementation of Cluster that uses a set of VMs listed in an ansible
+    inventory file. Hosts will be taken from the inventory group name
+    `node_group`. See tests/cluster/check_ansible.py for example inventory file.
+    """
+
+    def __init__(self, inv_filename: str, cluster_file="cluster.json",
+                 node_group="kafka"):
+
+        self.ansible_file = inv_filename
+        self.cluster_file = cluster_file
+        self.node_group = node_group
+
+        self.log = logging.getLogger(__file__)
+        self.log.setLevel(logging.DEBUG)
+
+        # translate to known json format
+        loader = DataLoader()
+        self.inventory = InventoryManager(loader=loader, sources=inv_filename)
+
+        cluster_json = {"nodes": self._get_nodes_from_ansible()}
+        super().__init__(cluster_json)
+
+    def _get_nodes_from_ansible(self) -> list[dict]:
+        cluster_json: dict[str, Any] = {}
+        nodes = self.inventory.get_groups_dict()
+
+        node_group: Optional[dict] = None
```

Comment:
an empty dictionary has a falsy value so you could do `node_group = {}` and wouldn't need the type annotations either

Reply:
Ah.. didn't think of that. Cool.